examples_test.go

```diff
@@ -0,0 +1,288 @@
+package eris_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/rotisserie/eris"
+)
+
+// todo: manually test how stack corrections work when using a "global" error value
+//       that wasn't created via eris.NewGlobal (e.g. the root error stack will be
+//       wrong and might cause issues when trying to insert wrap frames into it)
+
+// Demonstrates JSON formatting of wrapped errors that originate from
+// external (non-eris) error types.
+func ExampleUnpackedError_ToJSON_external() {
+	// example func that returns an IO error
+	readFile := func(fname string) error {
+		return io.ErrUnexpectedEOF
+	}
+
+	// unpack and print the error
+	err := readFile("example.json")
+	uerr := eris.Unpack(err)
+	format := eris.NewDefaultFormat(false) // false: omit stack trace
+	u, _ := json.Marshal(uerr.ToJSON(format))
+	fmt.Println(string(u))
+	// Output:
+	// {"external":"unexpected EOF"}
+}
+
+// Demonstrates JSON formatting of wrapped errors that originate from
+// global root errors (created via eris.NewGlobal).
+func ExampleUnpackedError_ToJSON_global() {
+	// declare a "global" error type
+	ErrUnexpectedEOF := eris.NewGlobal("unexpected EOF")
+
+	// example func that wraps a global error value
+	readFile := func(fname string) error {
+		return eris.Wrapf(ErrUnexpectedEOF, "error reading file '%v'", fname)
+	}
+
+	// example func that catches and returns an error without modification
+	parseFile := func(fname string) error {
+		// read the file
+		err := readFile(fname)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// unpack and print the error via uerr.ToJSON(...)
+	err := parseFile("example.json")
+	uerr := eris.Unpack(err)
+	format := eris.NewDefaultFormat(true) // true: include stack trace
+	u, _ := json.MarshalIndent(uerr.ToJSON(format), "", "\t")
+	fmt.Printf("%v\n", string(u))
+
+	// example output:
+	// {
+	//   "root": {
+	//     "message": "unexpected EOF",
+	//     "stack": [
+	//       "eris_test.ExampleUnpackedError_ToJSON_global.func1: /Users/morningvera/go/src/github.com/rotisserie/eris/examples_test.go: 41",
```

Comment:
The line numbers in trace cannot be traced back to original code so I suggest to add commented line numbers or denote code lines somehow

Reply:
I added commented line nums, how does it look?

Reply:
finished improving the look of example output

Reply:
It is better than previous but still confusing. At this point, I can't think of anything which can make it better so let's leave it be and address if someone comments on it.